cybersyn/scripts/layout.lua

```diff
@@ -97,7 +97,13 @@ function set_train_layout(map_data, train)
 			item_slot_capacity = item_slot_capacity + #inv
 		elseif carriage.type == "fluid-wagon" then
 			layout[#layout + 1] = 2
-			fluid_capacity = fluid_capacity + carriage.prototype.fluid_capacity
+
+			--This is absolutely the only way of knowing how big a fluid wagon is with quality right now
+			local oldfluid = carriage.get_fluid(1)
+			carriage.clear_fluid_inside()
+			local fluidsize = carriage.insert_fluid({name="water", amount=1e10})
```

Comment:
It would make sense to cache this information, at least for the current game session.